projects/plugins/jetpack/json-endpoints/jetpack/class.jetpack-json-api-plugins-modify-endpoint.php

```diff
@@ -185,8 +186,8 @@ public function callback( $path = '', $blog_id = 0, $object = null ) {
 	public function default_action() {
 		$args = $this->input();
 
-		if ( isset( $args['autoupdate'] ) && is_bool( $args['autoupdate'] ) ) {
-			if ( $args['autoupdate'] ) {
+		if ( ( isset( $args['autoupdate'] ) && is_bool( $args['autoupdate'] ) ) || $this->is_scheduled_updates ) {
+			if ( $args['autoupdate'] && ! $this->is_scheduled_updates ) {
```

Comment:
Reading this part of the code last week, I couldn't wrap my head around what this is supposed to do and why it's there. Can you help me understand how scheduled updates play a part in it and why it should only run when it's not a scheduled update?

These two lines seem to test whether is_scheduled_updates is truthy and then does something when it's falsy?

Reply:
> Can you help me understand how scheduled updates play a part in it and why it should only run when it's not a scheduled update?

Looking into the code more, this block should not run at all when we try to perform a scheduled update. 

> These two lines seem to test whether is_scheduled_updates is truthy and then does something when it's falsy?

Based on the description above, I've removed the unnecessary check.